src/cocotb/share/lib/vpi/VpiObj.cpp

```diff
@@ -63,48 +63,41 @@ int VpiArrayObjHdl::initialise(const std::string &name,
 
     /* After determining the range_idx, get the range and set the limits */
     vpiHandle iter = vpi_iterate(vpiRange, hdl);
-
-    s_vpi_value val;
-    val.format = vpiIntVal;
+    vpiHandle rangeHdl;
 
     if (iter != NULL) {
-        vpiHandle rangeHdl;
-        int idx = 0;
+        rangeHdl = vpi_scan(iter);
 
-        while ((rangeHdl = vpi_scan(iter)) != NULL) {
-            if (idx == range_idx) {
+#ifdef MODELSIM
```

Comment:
This code isn't Modelsim specific.
The problem in `recursive_discovery` test has to do with `dut._handle.get_handle_by_name(subpath)` as detailed in #3710.

Reply:
If it's not modelsim specific, why isn't there a testcase in the other sims that require this codepath? Modelsim was the only one that failed when I took this out, and taking this logic out sovled #3710 for me.

Reply:
I think the issue is that conceptually this code should *always* run. If we get a multidimensional array, such that `range_idx` is non-0, this code should always be run to get the correct range spec.

Perhaps the only reason that Questa hits this line is because Questa is the only sim that *tests* this line. Might this have to do with how pseudo-regions are handled?